anyway/parsers/news_flash_db_adapter.py

```diff
@@ -95,119 +95,10 @@ def get_source(self, news_flash_id):
         ).fetchone()
         return source[0]
 
-    def insert_new_flash_news(
-        self,
-        *,
-        accident,
-        source,
-        title,
-        link,
-        date,
-        author,
-        description,
-        location=None,
-        lat=None,
-        lon=None,
-        resolution=None,
-        region_hebrew=None,
-        district_hebrew=None,
-        yishuv_name=None,
-        street1_hebrew=None,
-        street2_hebrew=None,
-        non_urban_intersection_hebrew=None,
-        road1=None,
-        road2=None,
-        road_segment_name=None,
-        tweet_id=None,
-    ):
-        """
-        insert new news_flash to db
-        :param tweet_id: tweet_id if there is
-        :param region_hebrew: region - mahuz
-        :param district_hebrew: district - napa
-        :param yishuv_name: yishuv name
-        :param street1_hebrew: street1
-        :param street2_hebrew: street2
-        :param non_urban_intersection_hebrew: non urban intersection
-        :param road_segment_name: urban segment name
-        :param title: title of the news_flash
-        :param link: link to the news_flash
-        :param date: parsed date of the news_flash
-        :param author: author of the news_flash
-        :param description: description of the news flash
-        :param location: location of the news flash (textual)
-        :param lat: latitude
-        :param lon: longitude
-        :param road1: road 1 if found
-        :param road2: road 2 if found
-        :param resolution: resolution of found location
-        :param accident: is the news flash an accident
-        :param source: source of the news flash
-        """
-        temp = [
-            title,
-            link,
-            date,
-            author,
-            description,
-            location,
-            lat,
-            lon,
-            resolution,
-            region_hebrew,
-            district_hebrew,
-            yishuv_name,
-            street1_hebrew,
-            street2_hebrew,
-            non_urban_intersection_hebrew,
-            road1,
-            road2,
-            road_segment_name,
-            accident,
-            source,
-            tweet_id,
-        ]
-        title, link, date, author, description, location, lat, lon, resolution, region_hebrew, district_hebrew, yishuv_name, street1_hebrew, street2_hebrew, non_urban_intersection_hebrew, road1, road2, road_segment_name, accident, source, tweet_id = pd.Series(
-            temp
-        ).replace(
-            {pd.np.nan: None, "": None, 0: None, -1: None, " ": None}
```

Comment:
Note: I gave these coercions up.

Reply:
@Mano3 @elazarg not sure how giving up the coercions affects us.
Note that when querying the NewsFlash table, we rely on the fact that certain fields have a null value in db (for example, if road_segment_name *is null*, but road1 is *not null*, we know that we have a resolution of a road but not a segment. If suddenly the road_segment_name is an empty string this might brake our code, [for example here](https://github.com/hasadna/anyway/blob/dev/anyway/flask_app.py#L412).
Hence my thought when making sure null values in this table was that in these places data does not exist or not relevant (for example street when accident is in an interurban road). We can rethink this over, but it should be in a greater context of the way we'll use this table and decide which location field are relevant.

Reply:
I'd rather `assert` that we don't have nonsensical values. If they are indeed nonsensical.

Reply:
assert is a good solution, since every resolution is a defined list of required values.